src/main/java/com/epam/test_generator/state/machine/CaseStateMachineAdapter.java

```diff
@@ -0,0 +1,13 @@
+package com.epam.test_generator.state.machine;
+
+import com.epam.test_generator.entities.Case;
+import com.epam.test_generator.entities.Events;
+import com.epam.test_generator.entities.Status;
+import org.springframework.statemachine.annotation.WithStateMachine;
+import org.springframework.stereotype.Component;
+
+@Component
+@WithStateMachine
+public class CaseStateMachineAdapter extends AbstractStateMachineAdapter<Status, Events, Case> {
```

Comment:
Abstract classes are reasonable approach when writing library code or framework. In projects abstract classes are not welcome because abstractions add complexity for code reader.